openpype/tools/publisher/widgets/report_page.py

```diff
@@ -0,0 +1,1554 @@
+            return
+        self._update_needed = False
+
+        instance_ids = self._instance_ids_filter
+        to_hide = set()
+        if not instance_ids:
+            instance_ids = self._instances_by_id
+        else:
+            to_hide = set(self._instances_by_id) - set(instance_ids)
+
+        for instance_id in instance_ids:
+            widget = self._views_by_instance_id.get(instance_id)
+            if widget is None:
+                instance = self._instances_by_id[instance_id]
+                widget = InstanceLogsWidget(instance, self._content_widget)
+                widget.set_log_filters(self._visible_filters)
+                self._views_by_instance_id[instance_id] = widget
+                self._content_layout.addWidget(widget, 0)
+            else:
+                widget.setVisible(True)
+
+        for instance_id in to_hide:
+            widget = self._views_by_instance_id.get(instance_id)
+            if widget is not None:
+                widget.setVisible(False)
+
+    def _clear_widgets(self):
+        """Remove all widgets from layout and from cache."""
+
+        while self._content_layout.count():
+            item = self._content_layout.takeAt(0)
+            widget = item.widget()
+            if widget:
+                widget.setVisible(False)
+                widget.deleteLater()
+        self._views_by_instance_id = {}
+
+    def update_instances(self, instances):
+        """Update publish instance from report.
+
+        Args:
+            instances (list[_InstanceItem]): Instance data from report.
+        """
+
+        self._instances_order = [
+            instance.id for instance in instances
+        ]
+        self._instances_by_id = {
+            instance.id: instance
+            for instance in instances
+        }
+        self._instance_ids_filter = []
+        self._clear_needed = True
+        self._update_needed = True
+        self._update_instances()
+
+    def set_instances(self, instance_ids=None):
+        """Set instance filter.
+
+        Args:
+            instance_ids (Optional[list[str]]): List of instances to keep
+                visible. Pass empty list to hide all items.
+        """
+
+        self._instance_ids_filter = instance_ids
+        self._update_needed = True
+        self._update_instances()
+
+
+class CrashWidget(QtWidgets.QWidget):
+    """Widget shown when publishing crashes.
+
+    Contains only minimum information for artis with easy access to report
+    actions.
+    """
+
+    def __init__(self, controller, parent):
+        super(CrashWidget, self).__init__(parent)
+
+        main_label = QtWidgets.QLabel("This is not your fault", self)
+        main_label.setAlignment(QtCore.Qt.AlignCenter)
+        main_label.setObjectName("PublishCrashMainLabel")
+
+        report_label = QtWidgets.QLabel(
+            (
+                "Please report the error to your pipeline support"
+                " using one of the options below"
```

Comment:
Should this end with a dot?
```suggestion
                "Please report the error to your pipeline support"
                " using one of the options below."
```